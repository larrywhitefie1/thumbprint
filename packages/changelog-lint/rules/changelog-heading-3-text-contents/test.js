const test = require('../../test-utils');
const rule = require('./');

const valid = [
    '### Added',
    '### Changed',
    '### Deprecated',
    '### Removed',
    '### Fixed',
    '### Security',
];

const invalid = ['### Add', '### Foo'];

test({
    rule,
    valid,
    invalid,
});
