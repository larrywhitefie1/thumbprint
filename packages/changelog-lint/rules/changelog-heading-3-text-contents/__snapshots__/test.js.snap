// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invalid case #0 fails 1`] = `
Object {
  "input": "### Add",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Third level heading must say one of the following: “Added, Changed, Deprecated, Removed, Fixed, Security” (“Add” found).",
      "ruleId": "changelog-heading-3-text-contents",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;

exports[`invalid case #1 fails 1`] = `
Object {
  "input": "### Foo",
  "output": Array [
    Object {
      "column": 1,
      "fatal": false,
      "line": 1,
      "location": Object {
        "end": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "reason": "Third level heading must say one of the following: “Added, Changed, Deprecated, Removed, Fixed, Security” (“Foo” found).",
      "ruleId": "changelog-heading-3-text-contents",
      "source": "remark-lint",
      "stack": null,
    },
  ],
}
`;
